ziti-tunnel/docker/README.md

```diff
@@ -0,0 +1,194 @@
+  identity that ziti-tunnel will assume when connecting to the controller.
+- The "/netfoundry" directory must be mounted on its own volume.
+  - The one-time enrollment token "/netfoundry/${NF_REG_NAME}.jwt" must exist when
+    the container is started for the first time. This is the JWT that was downloaded
+    from the controller when the Ziti identity was created.
+  - "/netfoundry/${NF_REG_NAME}.json" is created when the identity is enrolled.
+    The "/netfoundry" volume must be persistent (that is, ${NF_REG_NAME}.json must
+    endure container restarts), since the enrollment token is only valid for one
+    enrollment.
+
+## Kubernetes
+
+The ziti-tunnel image can be used in Kubernetes either as a sidecar, which would
+intercept packets only from other containers in the pod definition or as a dedicated
+pod that sets `hostNetwork` to true.
+
+    $ cat app-with-ziti-tunnel-sidecar.yaml
+    apiVersion: v1
+    kind: PersistentVolumeClaim
+    metadata:
+      name: tunnel-sidecar-pv-claim
+    spec:
+      accessModes:
+        - ReadWriteOnce
+      resources:
+        requests:
+          storage: 1Gi
+    ---
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: ziti-tunnel-sidecar-demo
+    spec:
+      replicas: 1
+      selector:
+        matchLabels:
+          app: ziti-tunnel-sidecar-demo
+      strategy:
+        type: Recreate
+      template:
+        metadata:
+          labels:
+            app: ziti-tunnel-sidecar-demo
+        spec:
+          containers:
+          - image: centos
+            name: testclient
+            command: ["sh","-c","while true; set -x; do curl -sSLf ethzero.ziti.ui 2>&1; set +x; sleep 5; done"]
+          - image: netfoundry/ziti-tunnel:0.5.8-2554
+            name: ziti-tunnel
+            env:
+            - name: NF_REG_NAME
+              value: tunnel-sidecar
+            volumeMounts:
+            - name: tunnel-sidecar-jwt
+              mountPath: "/var/run/secrets/netfoundry.io/enrollment-token"
+              readOnly: true
+            - name: ziti-tunnel-persistent-storage
+              mountPath: /netfoundry
+            securityContext:
+              capabilities:
+              add:
+                - NET_ADMIN
+          dnsPolicy: "None"
+          dnsConfig:
+            nameservers:
+              - 127.0.0.1
+              - 8.8.8.8
+          restartPolicy: Always
+          volumes:
+          - name: ziti-tunnel-persistent-storage
+            persistentVolumeClaim:
+              claimName: tunnel-sidecar-pv-claim
+          - name: tunnel-sidecar-jwt
+            secret:
+              secretName: tunnel-sidecar.jwt
+
+
+## Azure IoT Hub
+
+This image can be used to run a module on an Azure IoT runtime.
+
+    # createOption.HostConfig for tun (instead of tproxy): \"Devices\":[{\"PathOnHost\":\"/dev/net/tun\",\"PathInContainer\":\"/dev/net/tun\",\"CgroupPermissions\":\"rwm\"}],
```

Comment:
these escaped quotes on purpose?

Reply:
this is intentional. `createOptions` is a "stringified JSON", so this could be copied/pasted into a manifest as-is. https://docs.microsoft.com/en-us/azure/iot-edge/module-edgeagent-edgehub